lib/finch/http1/pool.ex

```diff
@@ -116,34 +151,43 @@ defmodule Finch.HTTP1.Pool do
     :ok
   end
 
+  @impl Finch.Pool
+  def get_pool_status(finch_name, shp) do
+    PoolMetrics.get_pool_status(finch_name, shp)
+  end
+
   @impl NimblePool
-  def init_pool({registry, shp, opts}) do
+  def init_pool({registry, shp, opts, pool_size, start_pool_metrics?}) do
+    if start_pool_metrics?, do: {:ok, _} = PoolMetrics.init(registry, shp, pool_size)
+
     # Register our pool with our module name as the key. This allows the caller
     # to determine the correct pool module to use to make the request
     {:ok, _} = Registry.register(registry, shp, __MODULE__)
-    {:ok, {shp, opts}}
+    {:ok, {registry, shp, opts}}
   end
 
   @impl NimblePool
-  def init_worker({{scheme, host, port}, opts} = pool_state) do
+  def init_worker({_name, {scheme, host, port}, opts} = pool_state) do
     {:ok, Conn.new(scheme, host, port, opts, self()), pool_state}
   end
 
   @impl NimblePool
-  def handle_checkout(:checkout, _, %{mint: nil} = conn, pool_state) do
+  def handle_checkout(:checkout, _, %{mint: nil} = conn, {name, shp, _opts} = pool_state) do
     idle_time = System.monotonic_time() - conn.last_checkin
+    PoolMetrics.add(name, shp, in_use_connections: 1, total_checkout_count: 1)
```

Comment:
I think if we stored the boolean `enable_pool_metrics?` in the pool state and check that before the `add/3` calls, it could be slightly faster in highly concurrent environments than the current method of checking if the pterm is present